src/Implement/UnitTestHttpListener.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2021, 鍾淯丞, 周杰仕, 林仁鴻. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <arpa/inet.h>
+#include <unistd.h>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "HttpListener.h"
+#include <iostream>
+class TestCallback{
+
+  public:
+    static void TestCallbackFunction(const int &fd){
+      TestCallback::test_callback = true;
+    }
+    static bool test_callback;
+};
+
+bool TestCallback::test_callback = false;
+
+void TestClient(){
```

Comment:
This can go in an anonymous namespace.

Reply:
Done

Reply:
This is still outside of the anonymous namespace.

Reply:
Done